src/vcpkg/commands.add.cpp

```diff
@@ -101,13 +101,27 @@ namespace vcpkg::Commands
                     return dep.name == spec.name && !dep.host &&
                            structurally_equal(spec.platform.value_or(PlatformExpression::Expr()), dep.platform);
                 });
-                const auto features = Util::fmap(spec.features.value_or({}), [](auto& feature) {
+                auto feature_names = spec.features.value_or({});
+                bool is_core = false;
+                Util::erase_if(feature_names, [&](const auto& feature_name) {
+                    if (feature_name == "core")
+                    {
+                        is_core = true;
+                        return true;
+                    }
+                    return false;
+                });
```

Comment:
```suggestion
                const auto original_feature_names_size = feature_names.size();
                Util::erase_remove(feature_names, "core");
                const bool is_core = original_feature_names_size != feature_names.size();
```

Reply:
As an aside we might want to handle "default" and "*"? If you want.

Reply:
I initially handled default, `parse_qualified_specifier` does not like "default":
```
➜  vcpkg git:(fix-stale-workflow) ✗ vcpkg add port 'qt[default]'        
<unknown>:1:11: error: 'default' is a reserved feature name
    on expression: qt[default]
                             ^
```
but `*` actually adds `*` to the list of features 😅 

In general we could verify if the ports and features exists before adding them to the manifest file.